File: src/types/task.ts
export interface Task {
  id: number;
  project: string;
  desc: string;
  external_id: string;
  start: string;
  end: string | null;
  reported: boolean;
  shortcut: number | null;
  duration: number;
  has_runing_tasks?: boolean;
  children?: Task[];
}

export interface Project {
  label: string;
  value: string;
}

export interface ExternalID {
  label: string;
  value: string;
}

export interface Summary {
  worked_today: number;
  worked_week: number;
  worked_month: number;
  goal_today: number;
  goal_week: number;
  is_running: boolean;
  pending_sync_tasks: number;
}

export interface BtnOption {
  label: string;
  value: string;
}

File: src-tauri/src/core/task.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct Task {
    pub id: i64,
    pub desc: String,
    pub start: String,
    pub end: Option<String>,
    pub reported: bool,
    pub external_id: Option<String>,
    pub project: String,
    pub duration: u64,
}

#[derive(Debug, Clone, Serialize)]
pub struct Summary {
    pub this_week: u64,
    pub today: u64,
    pub is_running: bool,
}
